Source/Core/Core/FreeLookConfig.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+// IMPORTANT: UI etc should modify FreeLook::g_Config. Free Look code should read
+// FreeLook::g_ActiveConfig. The reason for this is to get rid of race conditions etc when the
+// configuration changes in the middle of a frame. This is done by copying FreeLook::g_Config to
+// FreeLook::g_ActiveConfig at the start of every frame. No one should ever change members of
+// FreeLook::g_ActiveConfig directly.
+
+#pragma once
+
+namespace FreeLook
+{
+enum class ControlType : int
+{
+  SixAxis,
+  FPS,
+  Orbital
+};
+
+struct CameraConfig
+{
+  ControlType iControlType;
+};
+
+// NEVER inherit from this class.
+struct Config final
+{
+  Config();
+  void Refresh();
+
+  CameraConfig cameraConfig;
+  bool bEnabled;
+};
+
+extern Config g_Config;
```

Comment:
Globals should be avoided in new code. Consider making `g_config` static rather than global (which is slightly better) and `GetConfig` / `GetActiveConfig`.

This would also let you enforce the rule that nothing should modify the active config directly (by returning a const-ref from GetActiveConfig).

Reply:
This is a great idea.  I also cleaned up the comment